Automattic/woocommerce-payments
# Prevent UPE onboarding checkboxes getting unchecked

Fixes #5193 

#### Changes proposed in this Pull Request

##### Summary of the issue
On the UPE onboarding form, thats seen when we first enable UPE ( additional payment options ), there is a checkbox selector in the onboarding wizard. Selecting multiple options within a short span of 1.5 seconds, made only the first box selected at the end of it, and subsequently selected boxes, unchecked. 

**Issue ( Before ) :** 

![upe-onboarding-Screen Capture on 2022-11-25 at 13-58-35](https://user-images.githubusercontent.com/4162931/203935932-58cedace-6d54-409e-806f-a6196e9e061a.gif)

The issue was happening due to the complete `paymentMethodsState` being sent to the state updation hook which [expects two parameters](https://github.com/Automattic/woocommerce-payments/blob/ed3976f66ab2433c7184c629135f115209c7a782/client/additional-methods-setup/upe-preview-methods-selector/add-payment-methods-task.js#L40) - the changed payment method name, and its status (boolean) 

The PR corrects the call to the change handler and sends the specific parameters, instead of the complete updated state. 

**Result (after):** 
![after](https://user-images.githubusercontent.com/4162931/205429158-9f88393c-2c43-45fc-b2c9-88360088eefc.gif)

#### Testing instructions

1. Make sure UPE is disabled. Within `Payments > Settings` you should see the enable option. Please click that 

![image](https://user-images.githubusercontent.com/4162931/203929763-4a5397e4-6342-4292-8645-c93717143c25.png)

If UPE is already enabled, click the three dots to the top corner of the section , and disable it. and then enable to replicate the error.

2. Click on `Enable in your store`. It should open the `Add additional payment methods` wizard. 
3. Try selecting a few payment methods there very quickly. 
4. You should now see all of them remaining checked. None of them should get unchecked automatically
5. Uncheck one or two of them and click on `Add payment methods` and complete the wizard
6. On the settings page that loads, you should see the selection of UPE methods matching the final selection on the onboarding wizard earlier.

The PR passes all existing tests that seem adequate to me. Do guide me if anything more is needed. 

-------------------


**Post merge**
